packages/bundler/src/modules/BundleManager.ts

```diff
@@ -202,6 +202,13 @@ export class BundleManager {
       } catch (e: any) {
         debug('failed 2nd validation:', e.message)
         // failed validation. don't try anymore
+        // no active "penalization" (opsSeen was increased, but opsIncluded was not, so over time it hurts reputation)
+        // EREP-015: special case: if it is account/factory failure, then decreases paymaster's opsSeen)
+        if (paymaster != null && e instanceof RpcError && e.code === ValidationErrors.SimulateValidation &&
+          (e?.message.match(/FailedOpWithRevert\(\d+,"AA[21]/)) != null) {
```

Comment:
This is a very complex `if` condition. May I suggest that you create `handleFailed2ndValidation` and `isErrorCausedByPaymaster` functions to extract this code from a `createBundle` god-function?